guide/guide-states.rst

```diff
@@ -267,59 +493,138 @@ Qubit (two-level) systems
 Having spent a fair amount of time on basis states that represent harmonic oscillator states, we now move on to qubit, or two-level quantum systems (for example a spin-1/2). To create a state vector corresponding to a qubit system, we use the same :func:`qutip.states.basis`, or :func:`qutip.states.fock`, function with only two levels:
 
 
-.. ipython::
+.. testcode:: [states]
 
-    In [1]: spin = basis(2, 0)
+    spin = basis(2, 0)
 
 Now at this point one may ask how this state is different than that of a harmonic oscillator in the vacuum state truncated to two energy levels?
 
-.. ipython::
+.. testcode:: [states]
 
-    In [1]: vac = basis(2, 0)
+    vac = basis(2, 0)
 
 At this stage, there is no difference.  This should not be surprising as we called the exact same function twice.  The difference between the two comes from the action of the spin operators :func:`qutip.operators.sigmax`, :func:`qutip.operators.sigmay`, :func:`qutip.operators.sigmaz`, :func:`qutip.operators.sigmap`, and :func:`qutip.operators.sigmam` on these two-level states.  For example, if ``vac`` corresponds to the vacuum state of a harmonic oscillator, then, as we have already seen, we can use the raising operator to get the :math:`\left|1\right>` state:
 
-.. ipython::
+.. testcode:: [states]
+
+    print(vac)
+
+.. testoutput:: [states]
+    :options: +NORMALIZE_WHITESPACE
 
-    In [1]: vac
+    Quantum object: dims = [[2], [1]], shape = (2, 1), type = ket
+    Qobj data =
+    [[1.]
+     [0.]]
 
-.. ipython::
+.. testcode:: [states]
 
-    In [1]: c = create(2)
+    c = create(2)
 
-    In [2]: c * vac
+    print(c * vac)
 
+.. testoutput:: [states]
+    :options: +NORMALIZE_WHITESPACE
+
+    Quantum object: dims = [[2], [1]], shape = (2, 1), type = ket
+    Qobj data =
+    [[0.]
+     [1.]]
 
 For a spin system, the operator analogous to the raising operator is the sigma-plus operator :func:`qutip.operators.sigmap`.  Operating on the ``spin`` state gives:
 
-.. ipython::
+.. testcode:: [states]
+
+    print(spin)
+
+.. testoutput:: [states]
+    :options: +NORMALIZE_WHITESPACE
+
+    Quantum object: dims = [[2], [1]], shape = (2, 1), type = ket
+    Qobj data =
+    [[1.]
+     [0.]]
```

Comment:
Here and in the following lines, is there a way to distinguish input and output in the rendered version?